common/src/main/java/org/apache/apex/common/util/JarHelper.java

```diff
@@ -107,6 +115,73 @@ public String getJar(Class<?> jarClass)
     return jar;
   }
 
+  public String getJar(Class<?> jarClass)
+  {
+    return getJar(jarClass, true);
+  }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file.
+   * If the class is an independent file the method makes jar file from the folder that contains the class
+   * @param jarClass Class
+   * @param makeJarFromFolder True if the method should make jar from folder that contains the independent class
+   * @param addJarDependencies True if the method should include dependent jar files
+   * @return Set of names of the jar-files
+   */
+  public Set<String> getJars(Class<?> jarClass, boolean makeJarFromFolder, boolean addJarDependencies)
+  {
+    String jar = getJar(jarClass, makeJarFromFolder);
+    Set<String> set = new HashSet<>();
+    if (jar != null) {
+      set.add(jar);
+    }
+
+    if (addJarDependencies && (jar != null)) {
+      try {
+        getDependentJarsFromManifest(new JarFile(jar), set);
+      } catch (IOException ex) {
+        logger.warn("Cannot open Jar-file {}", jar);
+      }
+    }
+    return set;
+  }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file.
+   * If the class is an independent file the method makes jar file from the folder that contains the class
+   * @param jarClass Class
+   * @return Set of names of the jar-files
+   */
+  public Set<String> getJars(Class<?> jarClass)
+  {
+    return getJars(jarClass, true, true);
+  }
+
+  /**
+   * Adds dependent jar-files from manifest to the target list of jar-files
+   * @param jarFile Jar file
+   * @param set Set of target jar-files
+   * @throws IOException
+   */
+  public void getDependentJarsFromManifest(JarFile jarFile, Set<String> set) throws IOException
+  {
+    String value = jarFile.getManifest().getMainAttributes().getValue(APEX_DEPENDENCIES);
+    if (!StringUtils.isEmpty(value)) {
+      Path folderPath = Paths.get(jarFile.getName()).getParent();
+      for (String jar : value.split(",")) {
+        File file = folderPath.resolve(jar).toFile();
+        if (file.exists()) {
+          set.add(file.getPath());
+          logger.debug("Jar-file {} was added as a dependent jar", file.getPath());
+        } else {
+          logger.warn("Jar-file {} does not exist", file.getPath());
```

Comment:
It will be good to provide details regarding original Jar that has a missing dependency.